src/components/PopupWithForm.js

```diff
@@ -7,6 +7,7 @@ class PopupWithForm extends Popup {
       this._handleFormSubmit = handleFormSubmit;
       this._submitButton = this._popupElement.querySelector('.modal__button')
       this._formInputs = this._popupForm.querySelectorAll(".modal__input");
+      this._submitButtonText = this._submitButton.textContent;
     }
 
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```